python/tank_vendor/shotgun_authentication/interactive_authentication.py

```diff
@@ -366,18 +364,46 @@ def _renew_session(user, session_token, credentials_handler):
                 user.get_login(),
                 user.get_http_proxy()
             )
-        except AuthenticationError:
-            _renew_session_disabled = True
-            logger.debug("Authentication cancelled, disabling authentication.")
+        except AuthenticationCancelled:
+            _is_authentication_cancelled = True
+            logger.debug("Authentication cancelled")
             user.clear_saved_user(user.get_host())
             raise
 
         logger.debug("Login successful!")
 
         user.set_session_token(session_token)
-        session_cache.cache_session_data(
-            user.get_host(), user.get_login(), user.get_session_token()
-        )
+
+
+# Makes access to the thread count and executing logic based on it thread
+# safe.
+_renew_session_lock = threading.Lock()
+# Number of threads who are trying to renew the session.
+_renew_session_thread_count = 0
+
+
+def _renew_session(user, session_token, credentials_handler):
+    # One more renewer.
+    global _renew_session_lock
+    with _renew_session_lock:
+        global _renew_session_thread_count
+        _renew_session_thread_count = _renew_session_thread_count + 1
```

Comment:
FYI `_renew_session_thread_count += 1` works in python (but there is no ++ operator)